sos/cleaner/parsers/hostname_parser.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.cleaner.parsers import SoSCleanerParser
+from sos.cleaner.mappings.hostname_map import SoSHostnameMap
+
+
+class SoSHostnameParser(SoSCleanerParser):
+
+    name = 'Hostname Parser'
+    map_file_key = 'hostname_map'
+    prep_map_file = 'sos_commands/host/hostname'
+    regex_patterns = [
+        r'(((\b|_)[a-zA-Z0-9-\.]{1,200}\.[a-zA-Z]{1,63}\b))'
```

Comment:
I think we can be more precise here (cf. RFC 1123 and 952, or https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address/106223#106223 or https://stackoverflow.com/questions/11809631/fully-qualified-domain-name-validation/20204811#20204811), but for now it's good.